src/main/java/com/es/phoneshop/model/product/recentlyViewed/RecentlyViewedList.java

```diff
@@ -0,0 +1,45 @@
+package com.es.phoneshop.model.product.recentlyViewed;
+
+import com.es.phoneshop.model.product.Product;
+
+import java.util.ArrayDeque;
+import java.util.Deque;
+import java.util.Iterator;
+
+public class RecentlyViewedList {
+    private Deque<Product> lastViewed;
+
+    public RecentlyViewedList() {
+        this.lastViewed = new ArrayDeque<>();
+    }
+
+    public void addProduct(Product product, int amount) {
+        if (lastViewed.contains(product)) {
+            Product first = lastViewed.peek();
+            if (first.equals(product)) {
+                return;
+            }
+            Iterator<Product> iterator = lastViewed.iterator();
+            while (iterator.hasNext()) {
+                Product nextProduct = iterator.next();
+                if (nextProduct.equals(product)) {
+                    iterator.remove();
+                    lastViewed.addFirst(product);
+                    break;
+                }
+            }
+            return;
+        }
+        if (lastViewed.size() == amount) {
+            lastViewed.removeLast();
+            lastViewed.addFirst(product);
+            return;
+        } else {
+            lastViewed.addFirst(product);
+        }
+    }
+
+    public Deque<Product> getLastViewed() {
+        return lastViewed;
```

Comment:
Return defensive copy of lastViewed, so that the underlying deque cannot be directly modified from servlets and services